tests/writer_test.py

```diff
@@ -0,0 +1,1048 @@
+rememberToDownloadARealRemindersApp = 1;\\n}"'
+        )
+
+        # Write floating point coordinates
+        node = classes.GSNode(point(499.99, 512.01), classes.GSNode.OFFCURVE)
+        self.assertWritesValue(
+            node,
+            '"499.99 512.01 OFFCURVE"'
+        )
+
+
+    def test_write_guideline(self):
+        line = classes.GSGuideLine()
+        # http://docu.glyphsapp.com/#GSGuideLine
+        line.position = point(56, 45)
+        line.angle = 11.0
+        line.name = "italic angle"
+        # selected: not written
+        self.assertWrites(line, dedent("""\
+            {
+            angle = 11;
+            name = "italic angle";
+            position = "{56, 45}";
+            }
+        """))
+
+    def test_write_annotation(self):
+        annotation = classes.GSAnnotation()
+        # http://docu.glyphsapp.com/#gsannotation
+        annotation.position = point(12, 34)
+        annotation.type = classes.TEXT
+        annotation.text = "Look here"
+        annotation.angle = 123.5
+        annotation.width = 135
+        self.assertWrites(annotation, dedent("""\
+            {
+            angle = 123.5;
+            position = "{12, 34}";
+            text = "Look here";
+            type = 1;
+            width = 135;
+            }
+        """))
+
+    def test_write_hint(self):
+        hint = classes.GSHint()
+        # http://docu.glyphsapp.com/#gshint
+        # FIXME: (jany) understand how hints are stored
+        layer = classes.GSLayer()
+        path1 = classes.GSPath()
+        layer.paths.append(path1)
+        node1 = classes.GSNode(point(100, 100))
+        path1.nodes.append(node1)
+        hint.originNode = node1
+
+        # FIXME: (jany) implement the same official Python API as for `origin`
+        # path1.nodes.append(node2)
+        # node2 = classes.GSNode(point(200, 200))
+        # hint.targetNode = node2
+        hint.target = point(0, 1)
+
+        # node3 = classes.GSNode(point(300, 300))
+        # path1.nodes.append(node3)
+        # hint.otherNode1 = node3
+        hint.other1 = point(0, 2)
+
+        # path2 = classes.GSPath()
+        # node4 = classes.GSNode(point(400, 400))
+        # path2.nodes.append(node4)
+        # hint.otherNode2 = node4
+        hint.other2 = point(1, 0)
+
+        hint.type = classes.CORNER
+        hint.options = classes.TTROUND | classes.TRIPLE
+        hint.horizontal = True
+        # selected: not written
+        hint.name = "My favourite hint"
+        self.assertWrites(hint, dedent("""\
+            {
+            horizontal = 1;
+            origin = "{0, 0}";
+            target = "{0, 1}";
+            other1 = "{0, 2}";
+            other2 = "{1, 0}";
+            type = 16;
+            name = "My favourite hint";
+            options = 128;
+            }
+        """))
+
+        # FIXME: (jany) What about the undocumented scale & stem?
+        #   -> Add a test for that
```

Comment:
@schriftgestalt The [file format documentation](https://github.com/schriftgestalt/GlyphsSDK/blob/master/GlyphsFileFormat.md) mentions `stem` and `scale`, but I don't see them in the [Python API for GSHint](https://docu.glyphsapp.com/#gshint). How can I set those values using an official API?